argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py

```diff
@@ -167,13 +168,20 @@ async def _get_search_responses(
             "record": record,
             "query": text_query,
             "order": vector_query.order,
-            "max_results": limit,
+            "max_results": min(limit + offset, SEARCH_MAX_SIMILARITY_SEARCH_RESULT),
```

Comment:
Based on this name, won't the user expect to just limit similarity but actually they're limiting all queries?

Reply:
Got it.